compiler/rustc_resolve/src/ident.rs

```diff
@@ -677,14 +677,21 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                                         innermost_binding,
                                         binding,
                                     )
-                                    || flags.contains(Flags::MACRO_RULES)
+                                {
+                                    Some(AmbiguityKind::MacroRulesVsModularized)
+                                } else if flags.contains(Flags::MACRO_RULES)
                                         && innermost_flags.contains(Flags::MODULE)
                                         && !this.disambiguate_macro_rules_vs_modularized(
                                             binding,
                                             innermost_binding,
                                         )
                                 {
-                                    Some(AmbiguityKind::MacroRulesVsModularized)
+                                    // should be impossible because of visitation order in
+                                    // visit_scopes
+                                    //
+                                    // we visit all macro_rules scopes (e.g. textual scope macros)
+                                    // before we visit any modules (e.g. path-based scope macros)
+                                    span_bug!(orig_ident.span, "ambiguous scoped macro resolutions with path-based scope resolution as first candidate")
```

Comment:
Nit: could you wrap the string to line length limit? rustfmt doesn't enforce it for string literals, but should still maintain it when possible.